server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -197,12 +263,54 @@ public void writeTo(StreamOutput out) throws IOException {
         if (hasFallback) {
             new WriteableSetting(setting.fallbackSetting, type).writeTo(out);
         }
-        // We are using known types so don't need the parser
+        // Write a boolean specifying whether the parser is an instanceof writeable
+        boolean isParserWriteable = setting.parser instanceof Writeable;
+        out.writeBoolean(isParserWriteable);
+        if (isParserWriteable) {
+            writeParser(out, setting.parser);
+        }
         // We are not using validator
         // Write properties
         out.writeEnumSet(setting.getProperties());
     }
 
+    private void writeParser(StreamOutput out, Object parser) throws IOException {
+        switch (type) {
+            case Boolean:
+                ((BooleanParser) parser).writeTo(out);
+                break;
+            case Integer:
+                ((IntegerParser) parser).writeTo(out);
+                break;
+            case Long:
+                ((LongParser) parser).writeTo(out);
+                break;
+            case Float:
+                ((FloatParser) parser).writeTo(out);
+                break;
+            case Double:
+                ((DoubleParser) parser).writeTo(out);
+                break;
+            case TimeValue:
+                if (parser instanceof MinMaxTimeValueParser) {
+                    out.writeBoolean(true);
+                    ((MinMaxTimeValueParser) parser).writeTo(out);
+                } else if (parser instanceof MinTimeValueParser) {
+                    out.writeBoolean(true);
+                    ((MinTimeValueParser) parser).writeTo(out);
+                }
+                break;
+            case ByteSizeValue:
+                if (parser instanceof ByteSizeValueParser) {
+                    ((ByteSizeValueParser) parser).writeTo(out);
+                }
+                break;
```

Comment:
I think you're missing an else for `instanceof MemorySizeValueParser` here.

Reply:
ACK